source/math/FloatExtension.ooc

```diff
@@ -110,7 +110,7 @@ extend Float {
 				value /= 10.0f
 				radix *= 10.0f
 			}
-			while (Float absolute(value) < pow(10.0f, valueDigits-1)) {
+			while (Float absolute(value) - pow(10.0f, valueDigits-1) < -1.0E-6f) {
```

Comment:
Sorry I didn't think of this sooner, but you should probably use `Float epsilon` instead of `1.0E-6f` - I think that's what it's for, after all.